internal/ethapi/api.go

```diff
@@ -1375,10 +1375,11 @@ type SendTxArgs struct {
 	Nonce    *hexutil.Uint64 `json:"nonce"`
 	// We accept "data" and "input" for backwards-compatibility reasons. "input" is the
 	// newer name and should be preferred by clients.
-	Data            *hexutil.Bytes  `json:"data"`
-	Input           *hexutil.Bytes  `json:"input"`
-	L1RollupTxId    *hexutil.Uint64 `json:"l1RollupTxId,omitempty" rlp:"nil,?"`
-	L1MessageSender *common.Address `json:"l1MessageSender,omitempty" rlp:"nil,?"`
+	Data              *hexutil.Bytes           `json:"data"`
+	Input             *hexutil.Bytes           `json:"input"`
+	L1RollupTxId      *hexutil.Uint64          `json:"l1RollupTxId,omitempty" rlp:"nil,?"`
+	L1MessageSender   *common.Address          `json:"l1MessageSender,omitempty" rlp:"nil,?"`
+	SignatureHashType *types.SignatureHashType `json:"signatureHashType,omitempty" rlp:"nil,?"`
```

Comment:
This will have the same JSON / RLP encoding issue where `omitempty` results in incorrect parsing, right?

Reply:
I actually don't see this struct being RLP encoded anywhere and the other fields don't have the RLP encoding tag, so I'm removing it from this struct.

Reply:
I don't think we actually need to use `omitempty` here, what do you think?